### DISCUSSION TITLE
2020/10/23/read-network-addresses-in-procfs/

### DISCUSSION THREAD

**Discussion by giscus** [bric3/bric3.github.io#43] [Blog Comments]
# 2020/10/23/read-network-addresses-in-procfs/

Java mostly, and general tech

https://blog.arkey.fr/2020/10/23/read-network-addresses-in-procfs/

**Comment by kvaleriy** [comment]
I can see following output:
root@las:/home/vkayukov# more /proc/16888/net/route 
Iface   Destination     Gateway         Flags   RefCnt  Use     Metric  Mask            MTU     Window  IRTT                                                       
eth0    00000000        0101FEA9        0003    0       0       0       00000000        0       0       0                                                                               
eth0    0101FEA9        00000000        0005    0       0       0       FFFFFFFF        0       0       0                                                                               

Based on information you have shared there is no flags like 0003 or 0005.

**Comment by hengin** [comment]
Nice trick, this helped me out.

I'm guessing the IP address appears reversed because it's stored in network (big endian) byte order, but is read and printed as a 32-bit number on a machine with little endian native byte order.
